run

```diff
@@ -373,10 +379,13 @@ setup() {
      install_cuda_dependencies_package libcudnn8-dev cuda$short_cuda_version
   fi
 
-  # Make sure we do not install any cuda12 packages
-  install_cuda_dependencies_package libnvinfer-dev cuda$short_cuda_version
-  install_cuda_dependencies_package libnvinfer-plugin-dev cuda$short_cuda_version
-  install_cuda_dependencies_package libnvonnxparsers-dev cuda$short_cuda_version
+  # Check the current version of libnvinfer-bin
+  installed_libnvinferbin=$(dpkg --status libnvinfer-bin 2>/dev/null | grep -Po '^Version: \K[^-]*')
+  installed_libnvinferbin_version=${installed_libnvinferbin%.*}
```

Comment:
Could you comment on the behavior you were seeing that prompted this change? Did we have different component version installed by default?

Reply:
Isn't the CUDA version tied to the `nvinfer` package version? Not sure I understand why we want to strip the CUDA version from the package when it is a common field:
```
# apt list | grep nvinfer

WARNING: apt does not have a stable CLI interface. Use with caution in scripts.

libnvinfer-bin/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-dev/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-dispatch-dev/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-dispatch10/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-headers-dev/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-headers-plugin-dev/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-lean-dev/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-lean10/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-plugin-dev/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-plugin10/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-vc-plugin-dev/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer-vc-plugin10/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
libnvinfer10/now 10.3.0.26-1+cuda12.5 amd64 [installed,local]
```

Reply:
For context the previous implementation was done to make sure cuda 12 was not installed and only cuda 11. This is clearly not working for CUDA 12.2 vs 12.6. The new code will help.

Reply:
Got it, thanks for clarifying.